actions/create-and-push-image-manifests/action.yaml

```diff
@@ -0,0 +1,53 @@
+---
+description: Pulls image digests from GitHub and pushes manifests to GAR
+inputs:
+  full-image-name:
+    description: Image name to push the manifest for
+    required: true
+  tag:
+    description: Tag to create and push the manifest for
+    required: true
+  environment:
+    description: Environment to suffix the image digest
+    required: true
+name: Create and push image manifest
+
+runs:
+  using: composite
+  steps:
+    - name: Checkout
+      uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11
+    - name: Checkout
+      uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+      with:
+        repository: grafana/shared-workflows
+        ref: multiarch-docker-builds-using-sh-runners
+        path: shared-workflows
+    - name: Download digests
+      uses: actions/download-artifact@v4
+      with:
+        merge-multiple: true
+        path: /tmp/digests
+        pattern: digests-*-${{ inputs.tag }}-${{ inputs.environment }}
+    - name: Set up Docker Buildx
+      uses: docker/setup-buildx-action@v3
+    - id: meta
+      name: Docker meta
+      uses: docker/metadata-action@v5
+      with:
+        images: ${{ inputs.full-image-name }}
+        tags: ${{ inputs.tag }}
+    - id: login-to-gar
+      uses: grafana/shared-workflows/actions/login-to-gar@main
+      with:
+        environment: ${{ inputs.environment }}
+    - name: Create manifest list and push
+      working-directory: /tmp/digests
+      shell: bash
+      run: |
+        docker buildx imagetools create $(jq -cr '.tags | map("-t " + .) | join(" ")' <<< "$DOCKER_METADATA_OUTPUT_JSON") \
+          $(printf '${{ inputs.full-image-name }}@sha256:%s ' *)
```

Comment:
Two suggestions:

1. Do the `jq` thing on a separate line for a bit better readability, also please add a comment showing what it's transforming (`from: xxx to: yyy`)
2. Put `inputs.full-image-name` through an `env` block on this step. Just so that you can't [give it a value like `';) rm -rf /;` or whatever and do mean things to the machine](https://securitylab.github.com/research/github-actions-untrusted-input/). (Applies to the "Inspect image" step too and maybe others where input is handled similarly)